src/components/PostDetails.tsx

```diff
@@ -1,105 +1,56 @@
-  const [visible, setVisible] = useState(false);
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import * as commentActions from '../features/commentsSlice';
+
+export const PostDetails: React.FC = () => {
+  const { selectedPost } = useAppSelector(state => state.posts);
+  const {
+    comments,
+    hasError,
+    loaded,
+    visible,
+  } = useAppSelector(state => state.comments);
+  const dispatch = useAppDispatch();
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (selectedPost) {
+      dispatch(commentActions.init(selectedPost.id));
     }
-  };
-
-  useEffect(() => {
-    loadComments();
-  }, []);
+  }
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  useEffect(loadComments, [selectedPost?.id]);
 
   const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
+    if (selectedPost) {
+      const newComment = {
         name,
         email,
         body,
-        postId: post.id,
-      });
-
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+        postId: selectedPost?.id,
+      };
+
+      dispatch(commentActions.addComment(newComment));
+      dispatch(commentActions.actions.setComments(newComment));
     }
   };
 
-  const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    setComments(
-      currentComments => currentComments.filter(
-        comment => comment.id !== commentId,
-      ),
-    );
-
-    await commentsApi.deleteComment(commentId);
+  const deleteComment = (commentId: number) => {
+    dispatch(commentActions.deleteCom(commentId));
+    dispatch(commentActions.actions.delComment(commentId));
```

Comment:
You are dispatching an action to delete a comment two times in a row. You may want to check if that is intentional, as it could lead to unexpected state updates.